spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/QuestionAnswerAdvisor.java

```diff
@@ -151,4 +148,18 @@ protected Filter.Expression doGetFilterExpression(Map<String, Object> context) {
 
 	}
 
+	/**
+	 * Customize {@link SearchRequest} for each request. The returned
+	 * {@link SearchRequest} will be used in
+	 * {@link VectorStore#similaritySearch(SearchRequest)} to find similar documents.
+	 * @param searchRequest the original {@link SearchRequest}
+	 * @param request the {@link AdvisedRequest} representing the current request
+	 * @param context the shared data between advisors in the chain
+	 * @return the customized {@link SearchRequest}
+	 */
+	protected SearchRequest customizeSearchRequest(SearchRequest searchRequest, AdvisedRequest request,
+			Map<String, Object> context) {
```

Comment:
Why is it necessary to inherit and reimplement the implementation to return a SearchRequest method, even though QuestionAndAnswerAdvisor is already receiving a SearchRequest object through the constructor?